lib/chef/provisioning/fog_driver/providers/openstack.rb

```diff
@@ -11,6 +11,37 @@ def creator
         compute_options[:openstack_username]
       end
 
+      def create_winrm_transport(machine_spec, machine_options, server)
+        remote_host = if machine_spec.reference['use_private_ip_for_ssh']
+                        server.private_ip_address
+                      elsif !server.public_ip_address
+                        Chef::Log.warn("Server #{machine_spec.name} has no public ip address.  Using private ip '#{server.private_ip_address}'.  Set driver option 'use_private_ip_for_ssh' => true if this will always be the case ...")
+                        server.private_ip_address
+                      elsif server.public_ip_address
+                        server.public_ip_address
+                      else
+                        fail "Server #{server.id} has no private or public IP address!"
+                      end
+		Chef::Log::info("Connecting to server #{remote_host}")
+		
+        port = machine_spec.reference['winrm_port'] || 5985
```

Comment:
A couple of notes about this:

1.  I like that you're pulling the information out of the reference.  It is the right thing to do for many reasons
1.  Because the `reference` is just a hash, you should be doing `machine_spec.reference['winrm']['port']` (and `['winrm']['password'] and `['winrm']['username']` below).  This will make it easier to read and is more ruby-style
1.  Where does the reference get populated with these?  I would expect to see something around [here](https://github.com/chef/chef-provisioning-fog/blob/master/lib/chef/provisioning/fog_driver/driver.rb#L340-L342).  If you use a nested hash (like I recommend in item 2) then all you would need to do is add `winrm` to that hash.
1. You should add documentation to [here](https://github.com/chef/chef-provisioning-fog/blob/master/lib/chef/provisioning/fog_driver/driver.rb#L76) describing the new `winrm` key that can be passed and what subkeys it accepts (port, username, password).